### PR TITLE
Add graphql support

### PR DESCRIPTION
**Pull request by meziantou**

### CODE REVIEW

File: NGitLab.Tests/GraphQLTests.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                Variables =
+                {
+                    ["path"] = project.PathWithNamespace,
+                },
+            });
+
+            Assert.AreEqual("gid://gitlab/Project/" + project.Id, response.Project.Id);
+        }
+
+        private sealed class ProjectReponse
```
Comment (louis-z) [inline NGitLab.Tests/GraphQLTests.cs:55]: You're missing an 's': `ProjectResponse`

File: NGitLab.Mock/Clients/GitLabClient.cs [hunk 2]
```diff
@@ -37,6 +39,12 @@ public GitLabClient(ClientContext context)
 
         public IMergeRequestClient MergeRequests => new MergeRequestClient(Context);
 
+        public IGraphQLClient GraphQL
+        {
+            get => _overrideGraphQLClient ?? new GraphQLClient(Context);
```
Comment (louis-z) [inline NGitLab.Mock/Clients/GitLabClient.cs:44]: Couldn't we store the new instance if `_overrideGraphQLClient` was null?
`get => _overrideGraphQLClient ??= new GraphQLClient(Context);`

Comment (meziantou) [reply NGitLab.Mock/Clients/GitLabClient.cs:44]: We could do it for all similar properties. You can do it in another PR if you want :)

File: NGitLab.Tests/GraphQLTests.cs (hunk 1 above)
Comment (meziantou) [reply NGitLab.Tests/GraphQLTests.cs:55]: I'm not sure, there is only one project in the response

Comment (louis-z) [reply NGitLab.Tests/GraphQLTests.cs:55]: I'm not talking about singular vs plural forms, I meant you spelled the word `Reponse` wrong.
It should have been spelled Re**s**ponse.

Comment (meziantou) [reply NGitLab.Tests/GraphQLTests.cs:55]: I should use a spell checker...

Comment (meziantou) [reply NGitLab.Tests/GraphQLTests.cs:55]: Fixed in https://github.com/ubisoft/NGitLab/pull/178

**Pull request merged**